src/main/java/mate/jdbc/dao/CarDaoImpl.java

```diff
@@ -0,0 +1,221 @@
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement getAllByDriverStatement =
+                        connection.prepareStatement(getAllByDriverQuery)) {
+            getAllByDriverStatement.setLong(1, driverId);
+            ResultSet resultSet = getAllByDriverStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(getCarFromResultSet(resultSet));
+            }
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get cars by driverId " + driverId, e);
+        }
+        cars.forEach(c -> c.setDrivers(getDriversForCar(c.getId())));
+        return cars;
+    }
+
+    @Override
+    public Car update(Car car) {
+        String updateCarQuery = "UPDATE cars SET model = ?, manufacturer_id = ? "
+                + "WHERE id = ? AND is_deleted = FALSE;";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement updateCarStatement =
+                        connection.prepareStatement(updateCarQuery)) {
+            updateCarStatement.setString(1, car.getModel());
+            updateCarStatement.setLong(2, car.getManufacturer().getId());
+            updateCarStatement.setLong(3, car.getId());
+            updateCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car " + car, e);
+        }
+        deleteCarFromCarsDrivers(car.getId());
+        insertDrivers(car);
+        return car;
+    }
+
+    @Override
+    public boolean delete(Long id) {
+        String deleteCarQuery = "UPDATE cars SET is_deleted = TRUE WHERE id = ?;";
+        int numberOfDeletedRows = 0;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement deleteCarStatement =
+                        connection.prepareStatement(deleteCarQuery)) {
+            deleteCarStatement.setLong(1, id);
+            numberOfDeletedRows = deleteCarStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete the car by id " + id, e);
+        }
+        deleteCarFromCarsDrivers(id);
+        return numberOfDeletedRows != 0;
+    }
+
+    private void insertDrivers(Car car) {
+        String insertDriversQuery = "INSERT INTO cars_drivers (car_id, driver_id) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement insertDriversToCarStatement =
+                        connection.prepareStatement(insertDriversQuery)) {
+            insertDriversToCarStatement.setLong(1, car.getId());
+            for (Driver driver : car.getDrivers()) {
+                insertDriversToCarStatement.setLong(2, driver.getId());
+                insertDriversToCarStatement.executeUpdate();
+            }
```

Comment:
@AleksandrLts what about this one? Have you tried to implement this advice? Or talked about it somewhere else? would be nice if you left some comments here on why it was not done, 
I am not sure if we have it in our lectures, I think we do not, so I will not count it as a mistake, but in case you are still interested, you can check [this](https://www.baeldung.com/jdbc-batch-processing) out and try to fix it